src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -166,11 +209,290 @@ protected SnapshotResult<YugabyteDBOffsetContext> doExecute(ChangeEventSourceCon
+                YBTable table = tableIdToTable.get(tableId);
+
+                String tabletId = tableIdToTabletId.getValue();
+                YBPartition part = new YBPartition(tabletId);
+                
+                 // Check if snapshot is completed here, if it is, then break out of the loop
+                if (snapshotCompletedTablets.size() == tableToTabletForSnapshot.size()) {
+                    LOGGER.info("Snapshot completed for all the tablets");
+                    // Commit checkpoints for all tablets to make them ready for streaming changes
+                    for (Pair<String, String> entry : tableToTabletIds) {
+                      try {
+                        // Only checkpoint in case this is the first time snapshot has been 
+                        // completed, otherwise we will end up setting the checkpoint to 0,0 even
+                        // for tablets for which snapshot has been completed in a previous run
+                        // of the connector
+                        if (!snapshotCompletedPreviously.contains(entry.getValue())) {
+                          YBClientUtils.setCheckpoint(this.syncClient, 
+                                                      this.ybConnectorConfig.streamId(),
+                                                      entry.getKey() /* tableId */,
+                                                      entry.getValue() /* tabletId */,
+                                                      0 /* term */, 0 /* index */,
+                                                      true /* initialCheckpoint */, 
+                                                      true /* bootstrap */);
+                        }
+                      } catch (Exception e) {
+                        throw new DebeziumException(e);
+                      }
+                    }
+
+                    return SnapshotResult.completed(previousOffset);
+                }
+
+                // Skip the tablet if snapshot has already been taken for this tablet
+                if (snapshotCompletedTablets.contains(tabletId)) {
+                  continue;
+                }
+
+                OpId cp = previousOffset.snapshotLSN(tabletId);
+
+                LOGGER.debug("Going to fetch from checkpoint {} for tablet {} for table {}", 
+                            cp, tabletId, table.getName());
+
+                if (!context.isRunning()) {
+                  LOGGER.info("Connector has been stopped");
+                  break;
+                }
+                
+                GetChangesResponse resp = this.syncClient.getChangesCDCSDK(table, 
+                    ybConnectorConfig.streamId(), tabletId, cp.getTerm(), cp.getIndex(), cp.getKey(), 
+                    cp.getWrite_id(), cp.getTime(), schemaNeeded.get(tabletId));
+                
+                // Process the response
+                for (CdcService.CDCSDKProtoRecordPB record : 
```

Comment:
You should check for cancel in this loop similar to the streaming change. Otherwise the connector may not be cancellable for a long time.

Reply:
It is already there on line L336